GUI/src/gui_qt.py

```diff
@@ -0,0 +1,458 @@
+# Copyright (C) 2025 LEIDOS.
+
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not
+# use this file except in compliance with the License. You may obtain a copy of
+# the License at
+
+# http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations under
+# the License.
+
+import sys
+import os
+import subprocess
+import threading
+import shutil
+from datetime import datetime
+import yaml
+from PySide6.QtWidgets import (
+    QApplication, QMainWindow, QPushButton, QComboBox, QLabel, QTextEdit, QFileDialog, QVBoxLayout, QHBoxLayout, QToolBar, QDialog, QWidget, QDialogButtonBox
+)
+from PySide6.QtCore import Qt
+from multiprocessing import Process, Queue
+
+class FileSelectDialog(QDialog):
+    def __init__(self, title, files, parent=None):
+        super().__init__(parent)
+        self.setWindowTitle(title)
+        self.setModal(True)
+        layout = QVBoxLayout()
+        
+        self.combo = QComboBox()
+        self.combo.addItems(files)
+        layout.addWidget(self.combo)
+        
+        button_box = QDialogButtonBox(QDialogButtonBox.Ok | QDialogButtonBox.Cancel)
+        button_box.accepted.connect(self.accept)
+        button_box.rejected.connect(self.reject)
+        layout.addWidget(button_box)
+        
+        self.setLayout(layout)
+        
+    def get_selected(self):
+        return self.combo.currentText() if self.result() == QDialog.Accepted else None
+
+class SimulatorGUI(QMainWindow):
+    def __init__(self):
+        super().__init__()
+        self.setWindowTitle("CDA Simulator GUI")
+        self.setMinimumSize(600, 400)
+        
+        self.repo_folder = None
+        self.selected_folder = None
+        self.config_folder = None
+        self.start_script = None
+        self.stop_script = None
+        self.docker_compose_file = None
+        self.build_script = None
+        self.log_queue = Queue()  # For multiprocessing log messages
```

Comment:
Is there a reason to use a multiprocessing Queue here over just a standard python list with `append()` and `pop()` being the queue functions? Not sure it makes any difference, but curious if there's a reason to prefer the Queue class over a list-primitive.